### ISSUE TITLE
7.0 fails on Windows only (but not 6.x)

### ISSUE THREAD

**Issue by Symbitic** [cmake-js/cmake-js#286]
Running my project on cmake-js 7.0 produces this error:
```
[build-*cpp] LINK : fatal error LNK1181: cannot open input file 'node.lib' [D:\a\node-simpleble\node-simpleble\build\simpleble-node.vcxproj]
[build-*cpp]   Building Custom Rule D:/a/node-simpleble/node-simpleble/CMakeLists.txt
[build-*cpp]   wrapper.cpp
[build-*cpp]   Auto build dll exports
[build-*cpp]      Creating library D:/a/node-simpleble/node-simpleble/build/Release/simpleble.lib and object D:/a/node-simpleble/node-simpleble/build/Release/simpleble.exp
[build-*cpp] LINK : warning LNK4199: /DELAYLOAD:NODE.EXE ignored; no imports found from NODE.EXE [D:\a\node-simpleble\node-simpleble\build\simpleble-wrapper.vcxproj]
[build-*cpp]   simpleble-wrapper.vcxproj -> D:\a\node-simpleble\node-simpleble\build\Release\simpleble.dll
```

Yes, I made all the changes (`execute_process(COMMAND ${CMAKE_AR} /def:${CMAKE_JS_NODELIB_DEF} ...)`) required for v7.
It builds fine on Linux and macOS, but not Windows. It works as soon as I downgrade to 6.x

My project is here: https://github.com/Symbitic/node-simpleble

**Comment by Julusian** [comment]
Hmm interesting, I get a very different error when trying on windows 10 to build https://github.com/Symbitic/node-simpleble/commit/189ba6358aa6a4546a8bcdca604191e95cbda9db with cmake 7.

```
adapter.obj : error LNK2019: unresolved external symbol napi_create_bigint_uint64 referenced in function "public: __cde
cl <lambda_0377c1da0fb79b9c5cc83426a7797400>::operator()(class SimpleBLE::Safe::Peripheral)const " (??R<lambda_0377c1da
0fb79b9c5cc83426a7797400>@@QEBA@VPeripheral@Safe@SimpleBLE@@@Z) [C:\Code\node-simpleble\build\simpleble-node.vcxproj]
adapter.obj : error LNK2019: unresolved external symbol napi_get_value_bigint_uint64 referenced in function "public: st
atic class Napi::Value __cdecl AdapterWrapper::Address(class Napi::CallbackInfo const &)" (?Address@AdapterWrapper@@SA?
AVValue@Napi@@AEBVCallbackInfo@3@@Z) [C:\Code\node-simpleble\build\simpleble-node.vcxproj]
bindings.obj : error LNK2001: unresolved external symbol napi_get_value_bigint_uint64 [C:\Code\node-simpleble\build\sim
pleble-node.vcxproj]
peripheral.obj : error LNK2001: unresolved external symbol napi_get_value_bigint_uint64 [C:\Code\node-simpleble\build\s
impleble-node.vcxproj]
C:\Code\node-simpleble\build\Release\simpleble.node : fatal error LNK1120: 2 unresolved externals [C:\Code\node-simpleb
le\build\simpleble-node.vcxproj]
```

This is a bug in cmake-js that is an easy fix

What does the log look like before that point?
What version of the visual studio tools/compiler do you have installed? maybe this is something specific to an older/newer version?

**Comment by Julusian** [comment]
I just had a go at reproducing this with a unit test, and I was before I added that `execute_process` block to the cmake file. So it sounds like something caused that to not generate the node.lib file as expected.

**Comment by Julusian** [comment]
@Symbitic If you are referring to the error in this ci run https://github.com/Symbitic/node-simpleble/actions/runs/3352119393/jobs/5554040886, it looks like you didnt add that `execute_process(COMMAND ${CMAKE_AR} /def:${CMAKE_JS_NODELIB_DEF} ...)` block until the following commit which included reverting cmake-js to v6

**Comment by Symbitic** [comment]
@Julusian When I change the project name, I do indeed stop getting the node.lib error and start getting the `unresolved external symbol napi_create_bigint_uint64` error.

This bug also goes away as soon as I downgrade cmake-js to 6.3.2

**Comment by Julusian** [comment]
Could you provide an example commit of using cmake-js 7 where it isnt working?

**Comment by Symbitic** [comment]
```
git clone --recurse-submodules --branch simpleble-port https://github.com/Symbitic/webbluetooth
```

**Comment by Julusian** [comment]
I have finally gotten around to looking at this, and it looks like https://github.com/cmake-js/cmake-js/commit/1a6bce79d13a1eccfc3b4804ce85b0b63d4420e4 solves it. 
This is published in v7.1.0

**Comment by lukas2** [comment]
Just to add a variant of this problem for others that might come here via search engine: 

We noticed that our app stopped working with a white screen. The logged error was simply:
```
Uncaught Error: A dynamic link library (DLL) initialization routine failed.
```

Looking for differences my colleague noticed that we had updated `cmake-js` to `7.0.0`

The solution was to update to 7.1.1.

Thanks for creating the issue and thx to @Julusian and others for the fix
